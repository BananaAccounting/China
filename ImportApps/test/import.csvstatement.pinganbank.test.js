// Copyright [2015] [Banana.ch SA - Lugano Switzerland]
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//


// @id = ch.banana.apps.import.csvstatement.pinganbank.test
// @api = 1.0
// @pubdate = 2017-11-24
// @publisher = Banana.ch SA
// @description = <TEST PingAnBank.csv 平安银行>
// @task = app.command
// @doctype = *.*
// @docproperties =
// @outputformat = none
// @inputdataform = none
// @includejs = ../import.csvstatement.pinganbank.js
// @timeout = -1


/*

SUMMARY
-------
Javascript test
1. Open the .ac2 file
2. Execute the .js script
3. Save the report



virtual void addTestBegin(const QString& key, const QString& comment = QString());
virtual void addTestEnd();

virtual void addSection(const QString& key);
virtual void addSubSection(const QString& key);
virtual void addSubSubSection(const QString& key);

virtual void addComment(const QString& comment);
virtual void addInfo(const QString& key, const QString& value1, const QString& value2 = QString(), const QString& value3 = QString());
virtual void addFatalError(const QString& error);
virtual void addKeyValue(const QString& key, const QString& value, const QString& comment = QString());
virtual void addReport(const QString& key, QJSValue report, const QString& comment = QString());
virtual void addTable(const QString& key, QJSValue table, QStringList colXmlNames = QStringList(), const QString& comment = QString());

 **/

// Register test case to be executed
Test.registerTestCase(new ImportExcelCsvPingabankTest());

// Here we define the class, the name of the class is not important
function ImportExcelCsvPingabankTest() {}

// This method will be called at the beginning of the test case
ImportExcelCsvPingabankTest.prototype.initTestCase = function () {}

// This method will be called at the end of the test case
ImportExcelCsvPingabankTest.prototype.cleanupTestCase = function () {}

// This method will be called before every test method is executed
ImportExcelCsvPingabankTest.prototype.init = function () {}

// This method will be called after every test method is executed
ImportExcelCsvPingabankTest.prototype.cleanup = function () {}

ImportExcelCsvPingabankTest.prototype.testImport = function () {

	Test.logger.addKeyValue("key1", "test2");
	Test.logger.addComment("Test vImportExcelCsvPingabankTest");

	var file = Banana.IO.getLocalFile("file:script/testcases/PingAnBank.csv平安银行.csv")
		file.codecName = "GBK"; // Default is UTF-8
	var fileContent = file.read();
	if (file.errorString) {
		Test.logger.addComment(file.errorString);
		Test.logger.addComment("Test impossible to import");
		return;
	}
	//Banana.console.debug(fileContent);
	var tabSeparatedText = exec(fileContent);
	var report = createReportFromTabSeparatedText(tabSeparatedText);
	Test.logger.addReport("Report converted text", report);
}

function createReportFromTabSeparatedText(tabSeparatedText) {

	var report = Banana.Report.newReport("Report from tabSeparatedText");
	var table = report.addTable();
	var arrData = Banana.Converter.csvToArray(tabSeparatedText, '\t');
	if (!arrData) {
		Test.logger.addComment("Error no data in tabSeparatedText");
		return;
	}
	for (var i = 0; i < arrData.length; i++) {
		var tableRow = table.addRow();
		var values = arrData[i];
		if (values) {
			for (var j = 0; j < values.length; ++j) {
				tableRow.addCell(values[j]);
			}
		}
	}
	return report;
}

