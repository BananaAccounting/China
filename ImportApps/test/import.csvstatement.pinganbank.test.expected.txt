%%info_test_name{P:/L/GitHubDomenico/China/ImportApps/test/import.csvstatement.pinganbank.test.js}
%%info_test_app{Banana90}{9.0.0.171120}
%%info_test_os{Windows 10 (10.0)}
%%info_test_qt{5.9.3}
%%info_test_date{2017-11-27}{13:55:03.144}
\documentclass{report}
\usepackage{color}
\usepackage{adjustbox}
\begin{document}
\newenvironment{tablevalue}[2]
   {\textbf{Table: #1}\\\begin{adjustbox}{max width=\textwidth}\tabular{#2}}
   {\endtabular\end{adjustbox}}\newenvironment{reportvalue}[1]{\textbf{Report: #1}\\}{\\}
\newcommand{\info}[2]{\textit{#1: #2}\\}
\newcommand{\fatalerror}[1]{\textbf{\textcolor[rgb]{1,0,0}{Fatal error: #1}}\\}
\newcommand{\keyvalue}[2]{\textbf{Keyvalue: #1} #2\\}
\newcommand{\excltest}[1]{#1}}
%Begin test data
\begin{testcase}{ImportExcelCsvPingabankTest}
\begin{test}{testImport}
\keyvalue{key1}{test2}
%Test vImportExcelCsvPingabankTest
\begin{reportvalue}{Report converted text}
\begin{tabular}{ l  l  l  l  l }
Date  & Description  & Income  & Expenses  & ExternalReference \\
2016-11-27  &  & 30000  &  & BP0111475	 \\
2016-11-27  &  &  & 48  & BP0213731	 \\
2016-11-27  &  &  & 5  & EB0144955	 \\
2016-11-27  &  &  & 15000  & EB0144955	 \\
2016-11-27  &  &  & 5  & EB0144959	 \\
2016-11-27  &  &  & 15000  & EB0144959	 \\
2016-11-27  &  & 100000  &  & EB0146267	 \\
2016-11-27  &  & 673353.87  &  & EB0184788	 \\
2016-11-27  &  & 13300000  &  & EB0240714	 \\
2016-11-27  &  & 8309191.04  &  & EB0281431	 \\
2016-11-27  &  &  & 83  & EB0297069	 \\
2016-11-27  &  &  & 8300000  & EB0297069	 \\
2016-11-27  &  & 6000000  &  & EB0327788	 \\
2016-11-27  &  &  & 2  & EB0334585	 \\
2016-11-27  &  &  & 93.5  & EB0334585	 \\
2016-11-27  &  &  & 6000000  & EB0361883	 \\
2016-11-27  &  &  & 60  & EB0361883	 \\
2016-11-27  &  & 300000  &  & EB0365486	 \\
2016-11-27  &  & 341977.9  &  & EB0367990	 \\
2016-11-27  &  & 70020000  &  & EB0396997	 \\
2016-11-27  &  & 2000000  &  & EB0400629	 \\
2016-11-27  &  & 3740000  &  & EB0401821	 \\
2016-11-27  &  &  & 80  & EB0403670	 \\
2016-11-27  &  &  & 73757500  & EB0403670	 \\
2016-11-27  &  & 60000000  &  & EB0420476	 \\
2016-11-27  &  & 130000  &  & EB0437292	 \\
2016-11-27  &  &  & 80  & EB0456760	 \\
2016-11-27  &  &  & 73757500  & EB0456760	 \\
2016-11-27  &  &  & 7.5  & EB0461423	 \\
2016-11-27  &  &  & 500000  & EB0461423	 \\
2016-11-27  &  & 10000  &  & EB0462121	 \\
2016-11-27  &  &  & 20000  & EB0464770	 \\
2016-11-27  &  &  & 5  & EB0464770	 \\
2016-11-27  &  & 737140.25  &  & EB0484903	 \\
2016-11-27  &  &  & 737140.25  & EB0487465	 \\
2016-11-27  &  &  & 8  & EB0487465	 \\
2016-11-27  &  &  & 5  & EB0491593	 \\
2016-11-27  &  &  & 20000  & EB0491593	 \\
2016-11-27  &  &  & 2  & EB0498311	 \\
2016-11-27  &  &  & 724.54  & EB0498311	 \\
2016-11-27  &  &  & 2.5  & EB0516728	 \\
2016-11-27  &  &  & 381  & EB0516728	 \\
2016-11-27  &  &  & 2.5  & EB0516737	 \\
2016-11-27  &  &  & 423  & EB0516737	 \\
2016-11-27  &  &  & 2000000  & EB0582941	 \\
2016-11-27  &  &  & 20  & EB0582941	 \\
2016-11-27  &  &  & 2  & EB0616679	 \\
2016-11-27  &  &  & 2360.3  & EB0616679	 \\
2016-11-27  &  & 509045.96  &  & EB0628007	 \\
2016-11-27  &  &  & 509045.96  & EB0639267	 \\
2016-11-27  &  &  & 10  & EB0639267	 \\
2016-11-27  &  & 4000000  &  & EB0668215	 \\
2016-11-27  &  & 1127682.87  &  & EB0713096	 \\
2016-11-27  &  &  & 5  & EB0733837	 \\
2016-11-27  &  &  & 80000  & EB0733837	 \\
2016-11-27  &  &  & 4000000  & EB0735108	 \\
2016-11-27  &  &  & 40  & EB0735108	 \\
2016-11-27  &  &  & 1000  & EB0738807	 \\
2016-11-27  &  &  & 2.5  & EB0738807	 \\
2016-11-27  &  &  & 1749  & EB0744822	 \\
2016-11-27  &  &  & 2.5  & EB0744822	 \\
2016-11-27  &  &  & 3370  & EB0744833	 \\
2016-11-27  &  &  & 2.5  & EB0744833	 \\
2016-11-27  &  & 10000000  &  & EB0751065	 \\
2016-11-27  &  & 8100  &  & EB0753575	 \\
2016-11-27  &  &  & 17385.23  & EB0755881	 \\
2016-11-27  &  &  & 5  & EB0755881	 \\
2016-11-27  &  &  & 84629.58  & EB0755888	 \\
2016-11-27  &  &  & 4  & EB0755888	 \\
2016-11-27  &  &  & 1130000  & EB0756495	 \\
2016-11-27  &  &  & 11.3  & EB0756495	 \\
2016-11-27  &  & 60000  &  & EB0768105	 \\
2016-11-27  &  &  & 100  & EB0773448	 \\
2016-11-27  &  &  & 10100000  & EB0773448	 \\
2016-11-27  &  & 3350416.66  &  & EB0814705	 \\
2016-11-27  &  &  & 33.5  & EB0844546	 \\
2016-11-27  &  &  & 3350416.66  & EB0844546	 \\
\end{tabular}

\end{reportvalue}
\keyvalue{Result}{Passed}
%%info_test_time_elapsed{00:00:00.012}
\end{test}
%%info_test_time_elapsed{00:00:00.012}
\end{testcase}
%%info_test_time_elapsed{00:00:00.116}
\end{document}
